### PR TITLE
feat(auth): Paginate API keys listing

### LINKED ISSUE #2442: Pagination of the `/keys` ressources
Pagination of the `keys` resources: https://github.com/meilisearch/specifications/pull/138 in the `text/0085-api-keys.md` file 

### Pagination
- [x] Update test
- [ ] Add an offset/limit pagination on `GET - /keys` following [this spec](https://github.com/meilisearch/specifications/pull/138)
    - [x] Add `limit` (Default: `20`) and `offset`(Default: `0`) query parameters.
    - [x] Add `limit`, `offset` and `total` in the response body.

### PR DESCRIPTION
**Pull request by ManyTheFish**
- [x] Update tests
- [x] Use Pagination helpers to paginate API keys (thanks to @irevoire)

fixes #2442

### CODE REVIEW

File: meilisearch-http/src/routes/api_key.rs [hunk 1]
```diff
@@ -46,20 +47,21 @@ pub async fn create_api_key(
...
         let keys = auth_controller.list_keys()?;
-        let res: Vec<_> = keys
-            .into_iter()
-            .map(|k| KeyView::from_key(k, &auth_controller))
-            .collect();
-        Ok(res)
+        let page_view = paginate.auto_paginate_sized(
+            keys.into_iter()
+                .map(|k| KeyView::from_key(k, &auth_controller)),
+        );
```
Comment (irevoire) [inline meilisearch-http/src/routes/api_key.rs:57]: Pretty cool that rust is able to keep the impl of ExactSizeIterator after a map :astonished:

**Review by irevoire — DISMISSED** [review]
Looks perfect to me :ok_hand: 
Thank you

bors merge

**Review by irevoire — APPROVED** [review]
bors merge

File: meilisearch-http/src/routes/api_key.rs (hunk 1 above)
Comment (Kerollmops) [reply meilisearch-http/src/routes/api_key.rs:57]: Yup! [`ExactSizeIterator` is impl on `Map<I>` if `I: ExactSizeIterator`](https://doc.rust-lang.org/std/iter/struct.Map.html#impl-ExactSizeIterator).

**Pull request merged**